models/Image.js

```diff
@@ -8,7 +9,72 @@ var Image = new keystone.List('Image', {
   map: { name: 'description' }
 });
 
-var bucket = config['options']['gcs config']['bucket'];
+const bucket = config['options']['gcs config']['bucket'];
+
+const resizedOpts = [['desktop', 2000], ['tablet', 1200], ['mobile', 800], ['w400', 400], ['tiny', 150]];
+
+/**
+ *  Trigger `ResizeImage` Google Cloud Function.
+ *
+ *  @param {string} filename - uploaded file name
+ *  @returns {Promise} resolve with an array of message ids or reject with error
+ */
+function triggerResizeCloudFunction(filename) {
+  const topicName = 'resize-image';
+  const inputBucket = bucket;
+  const outputBucket = 'twreporter-multimedia';
+  const pubSubClient = new PubSub()
+
+  const promises = []
+
+  resizedOpts.forEach((resizedOpt) => {
+    let [resizeName, resizeWidth] = resizedOpt;
+    const promise = pubSubClient
+      .topic(topicName)
+      .publish(Buffer.from(filename), {
+        input_bucket: inputBucket,
+        output_bucket: outputBucket,
+        target: resizeName,
+        width: `${resizeWidth}`,
+      })
+    promises.push(promise);
+  })
+
+  return Promise.all(promises)
+}
+
+/**
+ *  Build resize metadata.
+ *
+ *  @param {string} filename - uploaded file name
+ *  @param {number} originWidth - image width
+ *  @param {number} originHeight - image height
+ *  @returns {Object}
+ */
+function getResizeMetadata(filename, originWidth, originHeight) {
+  const [filenameWithoutExt, ext] = filename.split('.');
+  const targets = {};
+
+  resizedOpts.forEach(resizedOpt => {
+    let [resizeName, resizeWidth] = resizedOpt;
+    let resizeHeight = 0;
+
+    if (resizeWidth < originWidth) {
+      resizeHeight = Math.round((resizeWidth / originWidth) * originHeight);
+    } else {
+      resizeWidth = originWidth;
+      resizeHeight = originHeight;
+    }
+
+    targets[resizeName] = {
+      url: `https://www.twreporter.org/images/${filenameWithoutExt}-${resizeName}.${ext}`,
```

Comment:
Parameterize `https://www.twreporter.org/images` route so that we can overwrite in staging with nginx relay to staging bucket.